### ISSUE TITLE
InvalidCharacterError

### ISSUE THREAD

**Issue by Mikolaj** [ghcjs/ghcjs#659]
On ghc-8.4 branch, there are problems with values of type Data.Text.Text. The following code, from https://github.com/LambdaHack/LambdaHack/blob/master/Game/LambdaHack/Client/UI/Frontend/Dom.hs#L84), but with the added assertion:

```
let div = ("div" :: Text)
    !_A = assert (div /= "") ()
divBlockRaw <- createElement doc div
```

fails not at the assertion, but at `createElement` with

```
ExceptionInLinkedThread ThreadId 13 JavaScript exception: InvalidCharacterError: Failed to execute 'createElement' on 'Document': The tag name provided 
('') is not a valid name.
```

The same code, with `Text` changed to `String`, works fine.

**Comment by luite** [comment]
I suspect this would also happen in 8.2 (unless you actually tested that and didn't see it), since the string changes happened in that version.

**Comment by luite** [comment]
Ok, I can reproduce this

**Comment by cdsmith** [comment]
Just mentioning, because I noticed this: the assert in the original bug should never fail regardless of `div`, since `_A1` is never forced.

**Comment by luite** [comment]
should be fixed by ghcjs/ghcjs-base@4560541fbab5b77ee34d0348ec04bb1dda8e5db3

**Comment by Mikolaj** [comment]
@cdsmith: the bang forces the assertion all right; just verified. 

@luite: the fix works and I see no further problems at this time. Thank you!
